Topic 3 nodejs npm and javascript modules

davidleonel opened the pull request.

davidleonel committed:
Topic 3: 1,2,3,4

Not done

davidleonel committed:
Topic 3: 1,2,3,4

1-Move into the topic folder and add a package.json file initializing
the package. Tip: use npm init.

2-Install browserify globally in your environment. Tip: npm install -g
browserify.

3-Create the same Track class as in the previous practice, but inside a
CommonJS module. Tip: use module.exports.

4-Create an Artist class inside a module and set it as a dependency on
the Track module. Tip: use require.

davidleonel committed:
Topic 3: minor changes

minor changes

davidleonel committed:
Topic 3: 5

Create a file called app.js using browserify. Add that script to an
index.html. Check that it works opening it in the browser.

I was able to create the file but for some reason when i try to
inplement the API, i get the following error on the chrome console,
right away: Uncaught ReferenceError: require is not defined
The app.js file is referenced as a script indide of the index.html file
and the app.js file contains the definition for the require...
HELP!!! :(

troyax commented on 03-Node-npm-modules/app.js:
```diff
@@ -0,0 +1,62 @@
+	};
+};
+
+Artist.prototype.set = function(attr, values){
```
This `set` method does not seems to be OK, try to create the set method as you've already created and add a new one to add all the awards that you expect to be added, this seems to be a mistake due to the awards thing. Right?

troyax replied:
Ohhh this is in the compiled js, I'lll take a look at the source, but seems to be the same mistake. Sorry.

troyax commented on 03-Node-npm-modules/artist.js:
```diff
@@ -0,0 +1,24 @@
+Artist.prototype.set = function(attr, values){
+
+	for (i = 0; i < values.length; i++){
+		this[attr].push(values[i]);
```
Yeap, the same here

troyax commented:
Try to add a `.gitignore` file to the root of your project, and ignore the `node_modules` folder. If this is not created and this folder is not ignored you'll be pushing all the libraries that are not needed to be pushed and I'm not able to check the code on the `track` file. Thanks!

davidleonel committed:
Topic 3: Fixed a few methods but still geting the same error

Fixed the set and setAwards methods from the artist class, created the
new app.js using the " browserify artist.js track.js -o app.js " using
the command line.
Still getting the same error when i open the index.html file on the
chrome console  Uncaught ReferenceError: require is not defined
api.js:1

davidleonel committed:
Topic 3: Updated the api.js and index.html

Updated the files so it can show the artist awards using jquery.

davidleonel committed:
Topic 3: 5,6,7,8 Completed

7- Add jQuery as a module.

8- Using jQuery show Artist awards.

davidleonel committed:
Added gitignore

--

davidleonel committed:
No changes

--

fercho8787 commented on 03-Node-npm-modules/api.js:
```diff
@@ -0,0 +1,20 @@
+
+showAwards = function () {
+    var $content = $(".content");
+    var $a = droid.get("artist").recognice();
```
This var does not contain a jQuery element so it should not start with `$`

troyax replied:
+1

troyax commented on 03-Node-npm-modules/artist.js:
```diff
@@ -0,0 +1,20 @@
+};
+
+Artist.prototype.set = function (attr, value) {
+    var temp = attr;
```
variable temp not needed

troyax commented on 03-Node-npm-modules/index.html:
```diff
@@ -0,0 +1,10 @@
+<body>
+	<div class="content"></div>
+	
+	<script src = "app.js" type="text/javascript"></script>
```
This should be `src="app.js"`

fercho8787 replied:
**Super minor:** remove extra spaces around `=`

fercho8787 commented on 03-Node-npm-modules/artist.js:
```diff
@@ -0,0 +1,20 @@
+
+Artist.prototype.recognice = function(){
+	var msg = this.name + " has the following awards: ";
+	for(i = 0; i < this.awards.length; i++){
```
You need to declare `i` before use it.

troyax commented on 03-Node-npm-modules/track.js:
```diff
@@ -0,0 +1,19 @@
+};
+	
+Track.prototype.set = function(attr, value){
+	var temp = attr;
```
Same here with `temp` variable

fercho8787 replied:
This var `temp` is not necessary. You can simply do `this[attr] = value;`

fercho8787 commented on 03-Node-npm-modules/artist.js:
```diff
@@ -0,0 +1,20 @@
+Artist.prototype.recognice = function(){
+	var msg = this.name + " has the following awards: ";
+	for(i = 0; i < this.awards.length; i++){
+		msg += this.awards[i] + ' ';
```
This is good, but you can archive the same with the javascript's method `.join()`. Example `var msg = this.name + " has the following awards: " + this.awards.join(' ')`

davidleonel committed:
Topic 3: Fixed all the mistakes pointed out.

All fixed :)

troyax commented on 03-Node-npm-modules/api.js:
```diff
@@ -0,0 +1,20 @@
+    $content.append(a);  
+};
+
+$(document).on("ready", showAwards);
```
String into js files must be defined with single quotes

troyax commented on 03-Node-npm-modules/artist.js:
```diff
@@ -0,0 +1,19 @@
+Artist.prototype.recognice = function(){
+	var msg = this.name + " has the following awards: ";
+	
+	for(var i = 0; i < this.awards.length; i++){
```
variable definitions must be on upper part of the block, also try to use more detailed names for indexes variables, try to avoid the use of `i`, `j`, `z`

troyax commented on 03-Node-npm-modules/artist.js:
```diff
@@ -0,0 +1,19 @@
+	for(var i = 0; i < this.awards.length; i++){
+		msg += this.awards[i] + ' ';
+	};
+    return msg;
```
use `message` instead of `msg`, more characters are not an issue and message is more detailed than msg

troyax commented:
Done
